src/events.rs

```diff
@@ -12,60 +10,147 @@ use std::convert::Into;
 /// adopt the same scheme as serde_json's Value: https://github.com/serde-rs/json/blob/master/json/src/value.rs
 #[derive(Clone, PartialEq)]
 pub enum Value {
-    Json(String)
+    /// Represents a JSON Boolean
+    Bool(bool),
+
+    /// Represents a JSON signed integer
+    I64(i64),
+
+    /// Represents a JSON unsigned integer
+    U64(u64),
+
+    /// Represents a JSON floating point number
+    F64(f64),
+
+    /// Represents a JSON string
+    String(String),
 }
 
 impl Value {
     // JSON serialization belongs elsewhere, but keeps
     // the distinction between regular string and JSON data clear.
-    pub fn to_json<'a>(&'a self) -> &'a str {
-        match self {
-            &Value::Json(ref s) => &s
+    pub fn to_json<'a>(&'a self) -> String {
+        match *self {
+            Value::Bool(ref b) => b.to_string(),
+            Value::I64(ref n) => n.to_string(),
+            Value::U64(ref n) => n.to_string(),
+            Value::F64(ref n) => n.to_string(),
+            Value::String(ref s) => {
+                let mut quoted = String::with_capacity(s.len() + 2);
+                quoted.push('"');
+                quoted.push_str(s);
```

Comment:
Probably need to raise a ticket to port something like: https://github.com/serilog/serilog/blob/dev/src/Serilog/Formatting/Json/JsonValueFormatter.cs#L269